go/vt/discovery/keyspace_events.go

```diff
@@ -179,11 +170,19 @@ type shardState struct {
 	currentPrimary       *topodatapb.TabletAlias
 }
 
-// Subscribe returns a channel that will receive any KeyspaceEvents for all keyspaces in the current cell
+// Subscribe returns a channel that will receive any KeyspaceEvents for all keyspaces in the
+// current cell.
 func (kew *KeyspaceEventWatcher) Subscribe() chan *KeyspaceEvent {
 	kew.subsMu.Lock()
 	defer kew.subsMu.Unlock()
-	c := make(chan *KeyspaceEvent, 2)
+	// Use a decent size buffer to:
+	// 1. Avoid blocking the KEW
+	// 2. While not losing/missing any events
+	// 3. And processing them in the order received
+	// TODO: do we care about intermediate events?
+	// If not, then we could instead e.g. pull the first/oldest event
+	// from the channel, discard it, and add the current/latest.
+	c := make(chan *KeyspaceEvent, 10)
```

Comment:
How did you come up with 10? If there are m shards in the source and n shards in the target, how many events do we receive?

Reply:
> How did you come up with 10?

It's a nice round number, it should be more than enough to avoid blocking, it's used elsewhere for similar event watching channels:
```
❯ git grep 'make(chan' | grep '10)'
go/cmd/vtclient/cli/vtclient.go: seqChan = make(chan int, 10)
go/vt/throttler/max_replication_lag_module.go:          lagRecords:         make(chan replicationLagRecord, 10),
go/vt/throttler/throttler.go:   rateUpdateChan := make(chan struct{}, 10)
go/vt/topo/consultopo/watch.go: notifications := make(chan *topo.WatchData, 10)
go/vt/topo/etcd2topo/watch.go:  notifications := make(chan *topo.WatchData, 10)
go/vt/topo/etcd2topo/watch.go:  notifications := make(chan *topo.WatchDataRecursive, 10)
go/vt/topo/shard.go:    changes := make(chan *WatchShardData, 10)
go/vt/topo/srv_keyspace.go:     changes := make(chan *WatchSrvKeyspaceData, 10)
go/vt/topo/srv_vschema.go:      changes := make(chan *WatchSrvVSchemaData, 10)
go/vt/topo/zk2topo/watch.go:    c := make(chan *topo.WatchData, 10)
go/vt/vtctld/tablet_data_test.go:       ch = make(chan *querypb.StreamHealthResponse, 10)
go/vt/vtgate/engine/merge_sort.go:              row:    make(chan []sqltypes.Value, 10),
go/vt/vtgate/executor_select_test.go:   results := make(chan *sqltypes.Result, 10)
go/vt/vtorc/logic/vtorc.go:     snapshotDiscoveryKeys = make(chan string, 10)
go/vt/vttablet/sysloglogger/sysloglogger_test.go:       ch = make(chan *tabletenv.LogStats, 10)
go/vt/vttablet/sysloglogger/sysloglogger_test.go:       ch = make(chan *tabletenv.LogStats, 10)
go/vt/vttablet/sysloglogger/sysloglogger_test.go:       ch = make(chan *tabletenv.LogStats, 10)
go/vt/vttablet/sysloglogger/sysloglogger_test.go:       ch = make(chan *tabletenv.LogStats, 10)
go/vt/vttablet/tabletservermock/controller.go:          BroadcastData:       make(chan *BroadcastData, 10),
go/vt/vttablet/tabletservermock/controller.go:          StateChanges:        make(chan *StateChange, 10),
```

> If there are m shards in the source and n shards in the target, how many events do we receive?

The number of keyspace events a vtgate receives are not tied to the number of shards.